stglib/iq.py

```diff
@@ -320,53 +831,184 @@ def ds_add_attrs(ds):
+        ds["Roll_1217"].attrs["units"] = "degrees"
+    ds["Roll_1217"].attrs.update(
+        {
+            "long_name": "Instrument Roll",
+            "epic_code": "1217",
+        }
+    )
+    ds["Ptch_1216"].attrs.update(
+        {
+            "long_name": "Instrument Pitch",
+            "epic_code": "1216",
+        }
+    )
     ds["VbPercentGood"].attrs["long_name"] = "Vertical beam percent good"
     ds["HorizontalSkew"].attrs["long_name"] = "Horizontal skew"
-    ds["SystemInWater"].attrs[
-        "long_name"
-    ] = "Percentage of sample during which instrument was submerged (100% means it was submerged for entire sample)"
+    ds["SystemInWater"].attrs.update(
+        {
+            "long_name": "Percentage of sample during which instrument was submerged",
+            "note": "100% means it was submerged for entire sample",
+        }
+    )
 
     # Profile Variables
     for n in range(4):
-        ds["Profile_%d_Amp" % n].attrs["long_name"] = "Beam %d amplitude" % n
-        ds["Profile_%d_VelStd" % n].attrs["long_name"] = (
-            "Beam %d velocity profile standard deviation" % n
+        ds["Profile_AGC%d_122%d" % (n + 1, n + 1)].attrs.update(
+            {"units": "counts", "long_name": "Echo Intensity (AGC) beam %d" % (n + 1)}
         )
-        ds["Profile_%d_Vel" % n].attrs["long_name"] = "Beam %d velocity profile" % n
-
-    def add_attributes(var, dsattrs):
-        var.attrs.update(
-            {
-                "initial_instrument_height": dsattrs["initial_instrument_height"],
-                # 'nominal_instrument_depth': dsattrs['nominal_instrument_depth'],
-                "height_depth_units": "m",
-            }
+        ds["Profile_vel%d_%dStd" % (n + 1, n + 1277)].attrs[
+            "long_name"
+        ] = "beam %d velocity profile standard deviation" % (n + 1)
+        ds["Profile_vel%d_%d" % (n + 1, n + 1277)].attrs.update(
+            {"long_name": "beam %d current velocity" % (n + 1)}
         )
-
-    for var in ds.variables:
-        if (var not in ds.coords) and ("time" not in var):
-            add_attributes(ds[var], ds.attrs)
+        ds["Profile_blanking_distance%d" % (n + 1)].attrs.update(
+            {"long_name": "beam %d blanking distance" % (n + 1), "units": "m"}
+        )
+        ds["Profile_bin_size%d" % (n + 1)].attrs.update(
+            {"long_name": "beam %d bin size" % (n + 1), "units": "m"}
+        )
+        if "Profile_z" in ds:
+            ds["Profile_z%d" % (n + 1)].attrs.update(
+                {
+                    "standard_name": "height",
+                    "long_name": "beam %d bin height relative to %s"
+                    % ((n + 1), ds.attrs["geopotential_datum_name"]),
+                    "units": "m",
+                    "positive": "%s" % ds.attrs["orientation"],
+                    "axis": "Z",
+                }
+            )
 
     return ds
+
+
+# I think we can remove this?
```

Comment:
This might be called from a turnaround script? Not sure...

Reply:
Ok, we can leave this.